commons/zenoh-buffers/src/zslice.rs

```diff
@@ -137,13 +204,67 @@ impl ZSlice {
         self.len() == 0
     }
 
+    /// Checks if the underlying data is valid.
+    /// For shared memory buffers the underlying data may be forcibly deallocated, and
+    /// this method provides and interface to check it. For other underlying buffer types
+    /// this method always returns true
+    #[cfg(feature = "shared-memory")]
+    #[inline]
+    #[must_use]
+    pub fn is_valid(&self) -> bool {
```

Comment:
How can be guaranteed that a forced deallocation happens between the moment the user checks the buffer validity and the moment he accesses it?

Reply:
On read access - no guarantees, on write access there will be a locking mechanism added (not in place right now)